### PR TITLE
Add some temporary debugging to diagnose e2e test failures

### PR DESCRIPTION
**Pull request by pebrc**
* Print result of cat shards when health check tests fail.
* Set master service logging to trace to dianose cluster health issues.

### CODE REVIEW

File: test/e2e/test/elasticsearch/debug.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	"net/http"
+
+	"github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/client"
+)
+
+func printShards(clientFactory func() (client.Client, error)) func() {
+	return func() {
+		esClient, err := clientFactory()
+		if err != nil {
+			fmt.Printf("while creating es client %v", err)
```
Comment (anyasabo) [inline test/e2e/test/elasticsearch/debug.go:20]:
```suggestion
			fmt.Printf("error while creating es client: %v", err)
```

File: test/e2e/test/elasticsearch/debug.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+			return
+		}
+		catShards, err := http.NewRequest(
+			http.MethodGet,
+			"/_cat/shards",
```
Comment (anyasabo) [inline test/e2e/test/elasticsearch/debug.go:25]: might be worth adding a `?v` to include the header

File: test/e2e/test/elasticsearch/debug.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+			return
+		}
+		defer shards.Body.Close()
+		bytes, err := ioutil.ReadAll(shards.Body)
+		if err != nil {
+			fmt.Printf("error while reading response body: %v \n", err)
+			return
+		}
+		fmt.Println("GET _cat/shards")
+		fmt.Printf("%s \n", bytes)
```
Comment (sebgl) [inline test/e2e/test/elasticsearch/debug.go:44]: Should we also get `/_cluster/allocation/explain`?

Comment (pebrc) [reply test/e2e/test/elasticsearch/debug.go:44]: I would assume we have that already in the logs due to the trace logging, but I can add it

**Pull request merged**